eox_nelp/api_clients/pearson_engine.py

```diff
@@ -0,0 +1,144 @@
+        self.client_id = getattr(settings, "PEARSON_ENGINE_API_CLIENT_ID")
+        self.client_secret = getattr(settings, "PEARSON_ENGINE_API_CLIENT_SECRET")
+        self.authentication_path = "oauth2/token/"
+
+        super().__init__()
+
+    @property
+    def base_url(self):
+        """Return the base URL for Pearson Engine API."""
+        return getattr(settings, "PEARSON_ENGINE_API_URL")
+
+    def _get_user_data(self, user):
+        """
+        Retrieve user data for the request payload.
+
+        Args:
+            user: The user object containing user data.
+
+        Returns:
+            dict: The user data formatted for the request.
+        """
+        return {
+            "username": user.username,
+            "first_name": user.first_name,
+            "last_name": user.last_name,
+            "email": user.email,
+            "country": user.profile.country.code,
+            "city": user.profile.city,
+            "phone": user.profile.phone_number,
+            "address": user.profile.mailing_address,
+            "arabic_first_name": user.extrainfo.arabic_first_name,
+            "arabic_last_name": user.extrainfo.arabic_last_name,
+        }
+
+    def _get_platform_data(self):
+        """
+        Retrieve platform data for the request payload.
+
+        Returns:
+            dict: The platform data formatted for the request.
+        """
+        return {
+            "name": settings.PLATFORM_NAME,
+            "tenant": getattr(settings, "EDNX_TENANT_DOMAIN", None)
+        }
+
+    def _get_exam_data(self, exam_id):
+        """
+        Retrieve exam data for the request payload.
+
+        Args:
+            exam_id: The external key for the exam.
+
+        Returns:
+            dict: The exam data formatted for the request.
+        """
+        return {
+            "external_key": exam_id,
+        }
+
+    def import_candidate_demographics(self, user, **kwargs):
+        """
+        Import candidate demographics into Pearson Engine.
+
+        Args:
+            user: The user object containing user data.
+
+        Returns:
+            dict: The response from Pearson Engine API.
+        """
+        path = "rti/api/v1/candidate-demographics/"
+
+        candidate = {
+            "user_data": self._get_user_data(user),
+            "platform_data": self._get_platform_data(),
+            **kwargs
+        }
+
+        return self.make_post(path, candidate)
+
+    def import_exam_authorization(self, user, exam_id, **kwargs):
+        """
+        Import exam authorization data into Pearson Engine.
+
+        Args:
+            user: The user object containing user data.
+            exam_id: The external key for the exam.
+
+        Returns:
+            dict: The response from Pearson Engine API.
+        """
+        path = "rti/api/v1/exam-authorization/"
+        exam_data = {
+            "user_data": {"username": user.username},
+            "exam_data": self._get_exam_data(exam_id),
```

Comment:
I don't like that the inner key has the same name of dict. But it is only an opinion. I leave it to your consideration